arborist/server.go

```diff
@@ -102,7 +102,7 @@ func (server *Server) MakeRouter(out io.Writer) http.Handler {
 	router.HandleFunc("/health", server.handleHealth).Methods("GET")
 
 	router.Handle("/auth/mapping", http.HandlerFunc(server.handleAuthMappingGET)).Methods("GET")
-	router.Handle("/auth/mapping", http.HandlerFunc(server.parseJSON(server.handleAuthMappingPOST))).Methods("POST")
+	router.Handle("/auth/mapping", http.HandlerFunc(server.handleAuthMappingPOST)).Methods("POST")
```

Comment:
Is the main purpose of this change to bypass the `baseHandler` function call?

Reply:
The endpoint used to only accept input in the request body. Now that we allow passing a token and no body, we don't need `parseJSON` (it errors when there is no body). So i refactored the "body parsing" logic out of `parseJSON` so i could call it manually in `handleAuthMappingPOST()`, only when we need it. When only a token is passed, we don't call it

Reply:
that makes sense. so if this expression is satisfied 
`authHeader := r.Header.Get("Authorization"); authHeader != ""`
then we don't parse the body.